pinot-spi/src/main/java/org/apache/pinot/spi/data/readers/GenericRow.java

```diff
@@ -217,65 +217,11 @@ public boolean equals(Object obj) {
     }
     if (obj instanceof GenericRow) {
       GenericRow that = (GenericRow) obj;
-      if (!_nullValueFields.containsAll(that._nullValueFields) || !that._nullValueFields
```

Comment:
I'm not sure why #6525 expanded this, but two .containsAll should be the same as .equals

Reply:
make sense

Reply:
I think the issue was there could be a deep comparison for nested structure.

Reply:
you might have better insight than me then. would you prefer I leave it as as two `.containsAll`?

Reply:
I think you should be good if the test passed.